src/tests/Search.test.js

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import { Router, MemoryRouter } from 'react-router-dom';
+import { render, fireEvent } from '@testing-library/react';
+import { createMemoryHistory } from 'history';
+import '@testing-library/jest-dom/extend-expect';
+import App from '../App';
+import defaultSubreddit from '../helpers/constants';
+
+const renderSearchPage = (route) => (
+  render(
+    <MemoryRouter initialEntries={[route]}>
+      <App />
+    </MemoryRouter>,
+  )
+);
+
+describe('Search', () => {
+  it('input to be defaultSubreddit when clicking on Search link', () => {
+    const { getByRole } = renderSearchPage('/search/wholesomememes');
+    const searchInput = getByRole('textbox');
+    const headerSearchLink = getByRole('link', { name: /Search/ });
+
+    fireEvent.click(headerSearchLink);
+
+    expect(searchInput.value).toBe(defaultSubreddit);
+  });
+
+  it('input has subreddit from URL', () => {
+    const { getByRole } = renderSearchPage('/search/CasualKnitting');
+    const searchInput = getByRole('textbox');
+
+    expect(searchInput.value).toBe('CasualKnitting');
+  });
+
+  it('input changes when new subreddit is entered', () => {
+    const { getByRole } = renderSearchPage('/search');
+    const searchInput = getByRole('textbox');
+
+    fireEvent.change(searchInput, { target: { value: 'IllegallySmolCats' } });
+
+    expect(searchInput.value).toBe('IllegallySmolCats');
+  });
+
+  it('url changes when new subreddit is searched', () => {
+    const history = createMemoryHistory();
```

Comment:
You could use the same `renderSearchPage` function, but it's not very pretty. This is what they suggest in [the docs](https://reacttraining.com/react-router/web/guides/testing/checking-location-in-tests)

```
let history, location;
render(
  <MemoryRouter initialEntries={["/my/initial/route"]}>
    <App />
    <Route
      path="*"
      render={({ history, location }) => {
        history = history;
        location = location;
        return null;
      }}
    />
  </MemoryRouter>,
  node
  );
```

Not sure what `node` is. But I guess it's not required. Feel free to go either way :wink:

Reply:
Yeah, that isn't very pretty 😅